### PR TITLE
feat: allow safari alert with block/allow buttons to be handled

### PR DESCRIPTION
**Pull request by imurchie**
Add "Block"/"Allow" alerts to those recognized.
Simulator Screen Shot - appiumTest-C33F9E2A-D126-499E-B7EA-7EB124163596-iPhone X - 2020-03-24 at 09 08 28
Simulator Screen Shot - appiumTest-C33F9E2A-D126-499E-B7EA-7EB124163596-iPhone X - 2020-03-24 at 09 10 30
Simulator Screen Shot - appiumTest-C33F9E2A-D126-499E-B7EA-7EB124163596-iPhone X - 2020-03-24 at 09 09 49

This should fix https://github.com/appium/appium/issues/8582 and https://github.com/appium/appium/issues/14091

### CODE REVIEW

File: lib/commands/alert.js [hunk 1]
```diff
@@ -113,18 +113,21 @@ helpers.getAlert = async function getAlert () {
   }
 
   // determine that the name of the button is what is expected
```
Comment (mykola-mokhnach) [inline lib/commands/alert.js]: why cannot we handle this logic inside WDA? https://github.com/appium/WebDriverAgent/blob/f23062fab0228f2835d4324b2a814aa1649d115a/WebDriverAgentLib/Categories/XCUIApplication%2BFBAlert.m#L16 is the method. Just let me know if there are any other conditions to detect alerts and I'll modify the stuff correspondingly.

Comment (imurchie) [reply lib/commands/alert.js]: The WDA logic has _always_ failed for Safari alerts. All the info is in this function, so if you want to translate it over, great!

Comment (mykola-mokhnach) [reply lib/commands/alert.js]: Sure, I'll try to do that today. Would you assist me with testing?

Comment (mykola-mokhnach) [reply lib/commands/alert.js]: also, I'll need the page source output for the cases these alerts are visible, also on iPad

Comment (imurchie) [reply lib/commands/alert.js]: I can help with testing. There is also a test suite for Safari alerts within `appium-xcuitest-driver` (https://github.com/appium/appium-xcuitest-driver/blob/master/test/functional/web/safari-alerts-e2e-specs.js). A test for the popup is added in this PR.

Sources...
iPhone: https://gist.github.com/imurchie/421b17339a32e64f965e25c779fdd200
iPad: https://gist.github.com/imurchie/99043004e30f66ff2b1986bc808d0064

**Pull request merged**